scipy/integrate/_quadrature.py

```diff
@@ -822,6 +825,260 @@ def simpson(y, *, x=None, dx=1.0, axis=-1, even=_NoValue):
+    arr = np.asarray(arr)
+    if np.issubdtype(arr.dtype, np.integer):
+        arr = arr.astype(float, copy=False)
+    return arr
+
+
+def cumulative_simpson(y, *, x=None, dx=1.0, axis=-1, initial=None):
+    r"""
+    Cumulatively integrate y(x) using the composite Simpson's 1/3 rule.
+    The integral of the samples at every point is calculated by assuming a 
+    quadratic relationship between each point and the two adjacent points.
+
+    Parameters
+    ----------
+    y : array_like
+        Values to integrate.
+    x : array_like, optional
+        The coordinate to integrate along. Must be monotonically increasing.
+        If None (default), use spacing `dx` between consecutive elements in 
+        `y`.
+    dx : scalar or array_like, optional
+        Spacing between elements of `y`. Only used if `x` is None. Can either 
+        be a float, or an array with the same shape as `y`, but length 1 along
+        `axis`.
+    axis : int, optional
+        Specifies the axis to integrate along. Default is -1 (last axis).
+    initial : scalar or array_like, optional
+        If given, insert this value at the beginning of the returned result,
+        and add it to the rest of the result. Default is None, which means no
+        value at ``x[0]`` is returned and `res` has one element less than `y`
+        along the axis of integration. Can either be a float, or an array with
+        the same shape as `y`, but length 1 along `axis`.
+
+    Returns
+    -------
+    res : ndarray
+        The result of cumulative integration of `y` along `axis`.
+        If `initial` is None, the shape is such that the axis of integration
+        has one less value than `y`. If `initial` is given, the shape is equal
+        to that of `y`.
+
+    See Also
+    --------
+    numpy.cumsum
+    cumulative_trapezoid : cumulative integration using the composite 
+        trapezoidal rule
+    simpson : integrator for sampled data using the Composite Simpson's Rule
+
+    Notes
+    -----
+
+    .. versionadded:: 1.12.0
+
+    The composite Simpson's 1/3 method can be used to approximate the definite 
+    integral of a sampled input function :math:`y(x)` [1]_. The method assumes 
+    a quadratic relationship over the interval containing any 3 consecutive 
+    sampled points.
+
+    Consider three consecutive points: 
+    :math:`(x_1, y_1), (x_2, y_2), (x_3, y_3)`.
+    where the widths of the 2 subintervals are:
+    :math:`h_1 = x_2 - x_1` and :math:`h_2 = x_3 - x_2`.
+
+    Assuming a quadratic relationship over the 3 points, the integral over 
+    the subinterval between :math:`x_1` and :math:`x_2` is given by formula
+    (8) of [2]_:
+    
+    .. math::
+        \int_{x_1}^{x_2} y(x) dx\ = \frac{x_2-x_1}{6}\left[\
+        \left\{3-\frac{x_2-x_1}{x_3-x_1}\right\} y_1 + \
+        \left\{3 + \frac{(x_2-x_1)^2}{(x_3-x_2)(x_3-x_1)} + \
+        \frac{x_2-x_1}{x_3-x_1}\right\} y_2 - \
+        \frac{(x_2-x_1)^2}{(x_3-x_2)(x_3-x_1)} y_3\right]
```

Comment:
I think we need to line wrap

<img width="955" alt="image" src="https://github.com/scipy/scipy/assets/6570539/c4dacb81-4289-4a01-8c3f-3e741f4a9ef9">

Reply:
Added line wrap, but could not get it specifically aligned. Do you have an example of this elsewhere in the docs?
![image](https://github.com/scipy/scipy/assets/7060774/e0f15e6c-963f-4dc6-bc8b-10f16f48aa27)

Reply:
I'm not sure. This is fine with me. If someone else wants to fight with Sphinx, that's fine with me, but I'm happy as long as it doesn't extend past the right edge.